lib/appsignal/probes/erlang_probe.ex

```diff
@@ -74,19 +78,42 @@ defmodule Appsignal.Probes.ErlangProbe do
     set_gauge("total_run_queue_lengths", total - cpu, %{type: "io"})
   end
 
-  defp scheduler_utilization_metrics(sample) do
-    unless is_nil(sample) do
-      utilization = :scheduler.utilization(sample)
+  defp scheduler_utilization_metrics(nil, _), do: nil
 
-      utilization
-      |> Enum.map(&Tuple.to_list/1)
-      |> Enum.filter(fn [type | _] -> type == :normal end)
-      |> Enum.each(fn [_, id, value, _] ->
-        set_gauge("erlang_scheduler_utilization", value * 100, %{type: "normal", id: "#{id}"})
-      end)
+  defp scheduler_utilization_metrics(sample, next_sample) do
+    utilization = scheduler_utilization(sample, next_sample)
+
+    utilization
+    |> Enum.map(&Tuple.to_list/1)
+    |> Enum.filter(fn [type | _] -> type == :normal end)
+    |> Enum.each(fn [_, id, value, _] ->
+      set_gauge("erlang_scheduler_utilization", value * 100, %{type: "normal", id: "#{id}"})
+    end)
+  end
+
+  if System.otp_release() >= "21" do
```

Comment:
Is it possible to check if the `:scheduler` module is present here instead? Or is the module also present on OTP 20, but it doesn't have the feature we're using?

Reply:
Done! Thanks for the suggestion, it makes more sense this way.